packages/shared-ux/chrome/navigation/src/ui/components/navigation_section_ui.tsx

```diff
@@ -49,128 +129,109 @@ const navigationNodeToEuiItem = (
-    },
-    linkProps: { external: isExternal },
-    onClick:
-      href !== undefined
-        ? (event: React.MouseEvent) => {
-            event.preventDefault();
+  if (renderAs === 'panelOpener') {
+    const props: EuiCollapsibleNavSubItemProps = {
+      renderItem: () => <NavigationItemOpenPanel item={navNode} navigateToUrl={navigateToUrl} />,
+    };
+    return { props, isVisible: true };
+  }
+
+  const onClick = (e: React.MouseEvent) => {
+    if (href !== undefined) {
+      e.preventDefault();
+      navigateToUrl(href);
+      closePanel();
+      return;
+    }
+  };
+
+  const items: EuiCollapsibleNavItemProps['items'] = isItem
+    ? undefined
+    : navNode.children
+        ?.map((child) =>
+          nodeToEuiCollapsibleNavProps(child, {
+            navigateToUrl,
+            openPanel,
+            closePanel,
+            isSideNavCollapsed,
+            treeDepth: treeDepth + 1,
+          })
+        )
+        .filter(({ isVisible }) => isVisible)
+        .map((res) => res.props);
+
+  const linkProps: EuiCollapsibleNavItemProps['linkProps'] | undefined = hasLink
+    ? {
+        href,
+        external: isExternal,
+        onClick: (e: React.MouseEvent) => {
+          // TODO: here we might want to toggle the accordion (if we "renderAs: 'accordion'")
+          // Will be done in following PR
+          e.preventDefault();
+          e.stopPropagation();
+          if (href) {
             navigateToUrl(href);
           }
-        : undefined,
+        },
+      }
+    : undefined;
+
+  const accordionProps: Partial<EuiAccordionProps> | undefined = isItem
+    ? undefined
+    : {
+        initialIsOpen: true, // FIXME open state is controlled on component mount
+        ...navNode.accordionProps,
+      };
+
+  const props: EuiCollapsibleNavItemProps = {
+    id,
+    title,
+    isSelected,
+    accordionProps,
+    linkProps,
+    onClick,
     href,
-    items: item.children?.map((_item) =>
-      navigationNodeToEuiItem(_item, { navigateToUrl, basePath })
-    ),
+    items,
     ['data-test-subj']: dataTestSubj,
-    icon: item.icon,
-    iconProps: { size: 's' },
+    icon,
+    iconProps: { size: treeDepth === 0 ? 'm' : 's' },
   };
+
+  const isVisible = isItem || !_navNode.children || items !== undefined;
+  return { props, isVisible };
 };
 
 interface Props {
   navNode: ChromeProjectNavigationNode;
-  items?: ChromeProjectNavigationNode[];
 }
 
-export const NavigationSectionUI: FC<Props> = ({ navNode, items = [] }) => {
-  const { id, title, icon, isActive } = navNode;
-  const { navigateToUrl, basePath } = useServices();
-  const [isCollapsed, setIsCollapsed] = useState(!isActive);
-  // We want to auto expand the group automatically if the node is active (URL match)
-  // but once the user manually expand a group we don't want to close it afterward automatically.
-  const [doCollapseFromActiveState, setDoCollapseFromActiveState] = useState(true);
```

Comment:
For now all accordion are expanded by default so I removed this logic. I will put it back when working on https://github.com/elastic/kibana/issues/167323 and make it work to expand the accordion if it matches the URL location.